### ISSUE TITLE
Pyright does not correctly typecheck Literals per PEP 586

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#2584]
**Describe the bug**
Pyright does not correctly typecheck Literals per [PEP 586](https://www.python.org/dev/peps/pep-0586/#illegal-parameters-for-literal-at-type-check-time)

**To Reproduce**
Typecheck the following code block:

```python
from typing import Any, Literal, TypeVar
from pathlib import Path

Wrong1 = Literal[3 + 4]
Wrong2 = Literal["foo".replace("o", "b")]
Wrong3 = Literal[4 + 3j]
Wrong4 = Literal[-4 + 2j]
Wrong5 = Literal[(1, "foo", "bar")]
Wrong6 = Literal[{"a": "b", "c": "d"}] 
Wrong7 = Literal[Path("abcd")]
T = TypeVar("T")
Wrong8 = Literal[T]
Wrong9 = Literal[3.14]
Wrong10 = Literal[Any]
Wrong11 = Literal[...]
def func(): ...
Wrong12 = Literal[func]
some_variable = "foo"
Wrong13 = Literal[some_variable]
```

**Expected behavior**
I believe according to PEP 586 all instances of `Literal` in the snippet above should elicit an error.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

![image](https://user-images.githubusercontent.com/3920249/142681437-011b1db9-5cfc-4c5a-9826-0710bbea7bff.png)

**VS Code extension or command-line**
Both in VS Code and CLI using `"typeCheckingMode": "strict"`

**Comment by erictraut** [comment]
I agree that this is a false negative and these should be flagged as errors.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.189, which I just published. It will also be included in the next release of pylance.
